.github/workflows/tests.yml

```diff
@@ -0,0 +1,135 @@
+    name: ${{ matrix.target }} ${{ matrix.build_type }}
+    runs-on: ubuntu-latest
+    container: discourse/discourse_test:release
+    timeout-minutes: 60
+
+    env:
+      DISCOURSE_HOSTNAME: www.example.com
+      RUBY_GLOBAL_METHOD_CACHE_SIZE: 131072
+      RAILS_ENV: test
+      PGHOST: postgres
+      PGUSER: discourse
+      PGPASSWORD: discourse
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        build_type: [backend, frontend]
+        target: [core, plugins]
+        postgres: ["13"]
+        redis: ["6.x"]
+
+    services:
+      postgres:
+        image: postgres:${{ matrix.postgres }}
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_USER: discourse
+          POSTGRES_PASSWORD: discourse
+          POSTGRES_DB: discourse_test
+        options: >-
+          --mount type=tmpfs,destination=/var/lib/postgresql/data
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+
+      - name: Setup Git
+        run: |
+          git config --global user.email "ci@ci.invalid"
+          git config --global user.name "Discourse CI"
+
+      - name: Setup redis
+        uses: shogo82148/actions-setup-redis@v1
+        with:
+          redis-version: ${{ matrix.redis }}
+
+      - name: Bundler cache
+        uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gem-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gem-
+
+      - name: Setup gems
+        run: |
+          bundle config --local path vendor/bundle
+          bundle config --local deployment true
+          bundle config --local without development
+          bundle install --jobs 4
+          bundle clean
+
+      - name: Get yarn cache directory
+        id: yarn-cache-dir
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Yarn install
+        run: yarn install
+
+      - name: Checkout official plugins
+        if: matrix.target == 'plugins'
+        run: bin/rake plugin:install_all_official
+
+      - name: Create database
+        run: |
+          bin/rake db:create
+          bin/rake db:migrate
+
+      - name: Create parallel databases
+        if: matrix.build_type == 'backend' && matrix.target == 'core'
+        run: |
+          bin/rake parallel:create
+          bin/rake parallel:migrate
+
+      - name: Core RSpec
+        if: matrix.build_type == 'backend' && matrix.target == 'core'
+        run: |
+          bin/turbo_rspec
+          bin/rake plugin:spec
```

Comment:
Is it by design that plugin specs are run twice, once each for targets 'core' and 'plugins'?

Reply:
The first target runs core tests and the built-in plugin tests. The second one installs external plugins (`rake plugin:install_all_official`) and then runs tests on all plugins, including the core ones (which is a side effect; we could add some bash magic to skip bundled ones in that target)

Reply:
Or maybe we run baked-in-core plugin specs only in the second target alongside all official plugins?

Reply:
Sure, I can either way. The reasoning was that core tests catch things broken in discourse/discourse and plugin ones catch regressions in other repositories, but core/plugin (regardless of the source) division is fine too. I'll update the PR.